src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java

```diff
@@ -280,55 +280,84 @@ else if (tsDiff >= (-(1L << 30) & 0xFFFFFFFFL))
             int bMinSeqNum = b.getMinSeen(), aMaxSeqNum = a.getMaxSeen();
             int snDiff = (bMinSeqNum - aMaxSeqNum) & 0xFFFF;
 
-            if (bFirstSeqNum != -1 || aLastSeqNum != -1)
+            boolean guessed = false;
+
+            // We have some codecs (VPX) for which we can inspect the payload
+            // to deduce the start/end of a frame. This has an impact on the
+            // distances between the frames that allow us to guess frame
+            // boundaries.
+            boolean withFrameBoundaries =
+                a.supportsFrameBoundaries() && b.supportsFrameBoundaries();
+
+            if (withFrameBoundaries)
             {
-                if (snDiff == 2)
+                if (bFirstSeqNum != -1 || aLastSeqNum != -1)
                 {
-                    if (aLastSeqNum == -1)
+                    if (snDiff == 2 || snDiff == (-2 & 0xFFFF))
                     {
-                        aLastSeqNum = (aMaxSeqNum + 1) & 0xFFFF;
-                        if (logger.isDebugEnabled())
+                        if (aLastSeqNum == -1)
                         {
-                            logger.debug("Guessed frame end=" + aLastSeqNum);
+                            a.setEnd((aMaxSeqNum + 1) & 0xFFFF);
                         }
-                        a.setEnd(aLastSeqNum);
-                    }
-                    else
-                    {
-                        bFirstSeqNum = (bMinSeqNum - 1) & 0xFFFF;
-                        if (logger.isDebugEnabled())
+                        else
                         {
-                            logger.debug("Guessed frame start=" + bFirstSeqNum);
+                            b.setStart((bMinSeqNum - 1) & 0xFFFF);
                         }
-                        b.setStart(bFirstSeqNum);
+
+                        guessed = true;
                     }
                 }
-                else if (snDiff < 2 || snDiff > (-3 & 0xFFFF))
+                else // (bFirstSeqNum == -1 && aLastSeqNum == -1)
                 {
-                    logger.warn("Frame corruption or packets that are out of " +
-                        "order detected.");
+                    if (snDiff == 3 || snDiff == (-3 & 0xFFFF))
+                    {
+                        a.setEnd((aMaxSeqNum + 1) & 0xFFFF);
+                        b.setStart((bMinSeqNum - 1) & 0xFFFF);
+                        guessed = true;
+                    }
                 }
             }
             else
             {
-                if (snDiff == 3)
+                if (bFirstSeqNum != -1 || aLastSeqNum != -1)
                 {
-                    bFirstSeqNum = (bMinSeqNum - 1) & 0xFFFF;
-                    aLastSeqNum = (aMaxSeqNum + 1) & 0xFFFF;
-                    if (logger.isDebugEnabled())
+                    if (snDiff == 1 || snDiff == (-1 & 0xFFFF))
```

Comment:
You're right, if diff==1, then we know both the end of a and the start of b. It is also true that, if diff ==1, then we know that we must have seen the end (because the packet is marked), so `aLastSeqNum != -1`.

The reason why I prefer to keep the check is because of in-band probing. In that case a frame as identified by the RTP timestamp can be longer than the frame as identified by the marker bit and/or the end-of-frame indication inside the payload.